system/Database/BaseConnection.php

```diff
@@ -572,11 +583,11 @@ public function addTableAlias(string $table)
 	/**
 	 * Executes the query against the database.
 	 *
-	 * @param string $sql
+	 * @param mixed $sql
 	 *
 	 * @return mixed
 	 */
-	abstract protected function execute(string $sql);
+	abstract protected function execute($sql);
```

Comment:
As stated, changing the signature may be breaking for existing implementations.

Reply:
Does the var name `$sql` make sense in the context of NoSQL?

Reply:
The better name for the context will be `$statement`